libopenage/util/fixed_point.h

```diff
@@ -446,8 +447,37 @@ class FixedPoint {
 		return is;
 	}
 
+	/**
+	 * Pure FixedPoint sqrt implementation using Heron's Algorithm.
+	 *
+	 * Note that this function is undefined for negative values.
```

Comment:
Can you add an assert for this case? There is an `ENSURE` macro defined by openage that you can use for asserts.